### ISSUE TITLE
[Bug]: WatchGuardTechnologies.WatchGuardSystemManager publishing failed

### ISSUE THREAD

**Issue by NateHutch365** [svrooij/WingetIntune#30]
### Command

publish

### Full command

winget-intune publish WatchGuardTechnologies.WatchGuardSystemManager --package-folder C:\Winget

### App version

0.7.0+Branch.main.Sha.6f15dab446ea9f234d069287d6ca0c3eadbc54be.6f15dab446ea9f234d069287d6ca0c3eadbc54be

### What happened?

Expect it to open modern auth window to sign into 365 for publishing of app to intune

### Command output

```Shell
info: WingetIntune.Commands.PackageCommand[0] Try loading latest version from package index
info: WingetIntune.Implementations.WingetManager[5] Getting package info for WatchGuardTechnologies.WatchGuardSystemManager 12.10.1 from github
info: WingetIntune.Commands.PackageCommand[0] Publishing package WatchGuardTechnologies.WatchGuardSystemManager 12.10.1
fail: WingetIntune.IntuneManager[0] Error publishing app, deleting the remains System.InvalidOperationException: Sequence contains no elements    at System.Linq.ThrowHelper.ThrowNoElementsException()    at System.Linq.Enumerable.Aggregate[TSource](IEnumerable`1 source, Func`3 func)    at Microsoft.Kiota.Serialization.Json.JsonSerializationWriter.WriteEnumValue[T](String key, Nullable`1 value)    at Microsoft.Graph.Beta.Models.Win32LobApp.Serialize(ISerializationWriter writer)    at Microsoft.Kiota.Serialization.Json.JsonSerializationWriter.WriteObjectValue[T](String key, T value, IParsable[] additionalValuesToMerge)    at Microsoft.Kiota.Abstractions.RequestInformation.SetContentFromParsable[T](IRequestAdapter requestAdapter, String contentType, T item)    at Microsoft.Graph.Beta.DeviceAppManagement.MobileApps.MobileAppsRequestBuilder.ToPostRequestInformation(MobileApp body, Action`1 requestConfiguration)    at Microsoft.Graph.Beta.DeviceAppManagement.MobileApps.MobileAppsRequestBuilder.PostAsync(MobileApp body, Action`1 requestConfiguration, CancellationToken cancellationToken)    at WingetIntune.Graph.MobileAppsRequestBuilderExtensions.PostAsync(MobileAppsRequestBuilder builder, Win32LobApp win32LobApp, CancellationToken cancellationToken) in /home/runner/work/WingetIntune/WingetIntune/src/WingetIntune/Graph/MobileAppsRequestBuilderExtensions.cs:line 15    at WingetIntune.IntuneManager.PublishAppAsync(String packagesFolder, PackageInfo packageInfo, IntunePublishOptions options, CancellationToken cancellationToken) in /home/runner/work/WingetIntune/WingetIntune/src/WingetIntune/Intune/IntuneManager.cs:line 213
Unhandled exception: System.InvalidOperationException: Sequence contains no elements
   at System.Linq.ThrowHelper.ThrowNoElementsException()
   at System.Linq.Enumerable.Aggregate[TSource](IEnumerable`1 source, Func`3 func)
   at Microsoft.Kiota.Serialization.Json.JsonSerializationWriter.WriteEnumValue[T](String key, Nullable`1 value)
   at Microsoft.Graph.Beta.Models.Win32LobApp.Serialize(ISerializationWriter writer)
   at Microsoft.Kiota.Serialization.Json.JsonSerializationWriter.WriteObjectValue[T](String key, T value, IParsable[] additionalValuesToMerge)
   at Microsoft.Kiota.Abstractions.RequestInformation.SetContentFromParsable[T](IRequestAdapter requestAdapter, String contentType, T item)
   at Microsoft.Graph.Beta.DeviceAppManagement.MobileApps.MobileAppsRequestBuilder.ToPostRequestInformation(MobileApp body, Action`1 requestConfiguration)
   at Microsoft.Graph.Beta.DeviceAppManagement.MobileApps.MobileAppsRequestBuilder.PostAsync(MobileApp body, Action`1 requestConfiguration, CancellationToken cancellationToken)
   at WingetIntune.Graph.MobileAppsRequestBuilderExtensions.PostAsync(MobileAppsRequestBuilder builder, Win32LobApp win32LobApp, CancellationToken cancellationToken) in /home/runner/work/WingetIntune/WingetIntune/src/WingetIntune/Graph/MobileAppsRequestBuilderExtensions.cs:line 15
   at WingetIntune.IntuneManager.PublishAppAsync(String packagesFolder, PackageInfo packageInfo, IntunePublishOptions options, CancellationToken cancellationToken) in /home/runner/work/WingetIntune/WingetIntune/src/WingetIntune/Intune/IntuneManager.cs:line 213
   at WingetIntune.IntuneManager.PublishAppAsync(String packagesFolder, PackageInfo packageInfo, IntunePublishOptions options, CancellationToken cancellationToken) in /home/runner/work/WingetIntune/WingetIntune/src/WingetIntune/Intune/IntuneManager.cs:line 280
   at WingetIntune.Commands.PublishCommand.HandleCommand(PublishCommandOptions options, InvocationContext context) in /home/runner/work/WingetIntune/WingetIntune/src/WingetIntune.Cli/Commands/PublishCommand.cs:line 105
   at System.CommandLine.NamingConventionBinder.CommandHandler.GetExitCodeAsync(Object returnValue, InvocationContext context)
   at System.CommandLine.NamingConventionBinder.ModelBindingCommandHandler.InvokeAsync(InvocationContext context)
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass17_0.<<UseParseErrorReporting>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Hosting.HostingExtensions.<>c__DisplayClass1_0.<<UseHost>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass12_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass22_0.<<UseVersionOption>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass19_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__18_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass16_0.<<UseParseDirective>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<RegisterWithDotnetSuggest>b__5_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass8_0.<<UseExceptionHandler>b__0>d.MoveNext()
```

### Windows version and current language

Windows 11 Version 22H2 (OS Build 2261.3007)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I won't demand an immediate answer

**Comment by svrooij** [comment]
Can you try updating to this version and retry?

https://www.nuget.org/packages/SvRooij.Winget-Intune.Cli/0.8.0-alpha0008

I think this is the same issue as the one we had with some other software packages.

**Comment by NateHutch365** [comment]
Yep, confirmed that worked with following command

wintuner publish WatchGuardTechnologies.WatchGuardSystemManager --package-folder C:\Winget

**Comment by NateHutch365** [comment]
Sorry, it was a different WatchGuard app I published (which was also failing) but the alpha version resolved it for that (app id: WatchGuard.MobileVPNWithSSLClient) 

I just tried again with the WatchGuardTechnologies.WatchGuardSystemManager package and it failed with the below:

wintuner publish WatchGuardTechnologies.WatchGuardSystemManager --package-folder C:\Winget
info: WingetIntune.Commands.PackageCommand[0] Try loading latest version from package index
info: WingetIntune.Implementations.WingetManager[5] Getting package info for WatchGuardTechnologies.WatchGuardSystemManager 12.10.1 from github
info: WingetIntune.Commands.PackageCommand[0] Publishing package WatchGuardTechnologies.WatchGuardSystemManager 12.10.1
info: WingetIntune.IntuneManager[0] Created app da849ab5-f722-4c1e-8569-3b303a1ab4fa, starting with content
info: WingetIntune.Implementations.ChunkedAzureFileUploader[0] Uploading C:\Users\LocalAdmin\AppData\Local\Temp\e47071c5-2ded-46e3-9682-6be603a4380f\IntuneWinPackage\Contents\IntunePackage.intunewin to https://mmcswdb01.blob.core.windows.net/648ccf00-2e87-48d8-b498-9422802aafc3/f7de2986-0925-4ff6-bee8-9ed6eb71a367/a95b9276-1b6f-438f-ae7d-79186c58d5f1.intunewin.bin?sv=2017-04-17&sr=b&si=-132219447&sig=Is9LoH2hQXIAQYfU9bhwZjWun3pEZVIYuY%2BWGBgdZm4%3D
fail: WingetIntune.Implementations.ChunkedAzureFileUploader[0] Failed to upload C:\Users\LocalAdmin\AppData\Local\Temp\e47071c5-2ded-46e3-9682-6be603a4380f\IntuneWinPackage\Contents\IntunePackage.intunewin to https://mmcswdb01.blob.core.windows.net/648ccf00-2e87-48d8-b498-9422802aafc3/f7de2986-0925-4ff6-bee8-9ed6eb71a367/a95b9276-1b6f-438f-ae7d-79186c58d5f1.intunewin.bin?sv=2017-04-17&sr=b&si=-132219447&sig=Is9LoH2hQXIAQYfU9bhwZjWun3pEZVIYuY%2BWGBgdZm4%3D System.Threading.Tasks.TaskCanceledException: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.  ---> System.TimeoutException: The operation was canceled.  ---> System.Threading.Tasks.TaskCanceledException: The operation was canceled.  ---> System.Net.Http.HttpRequestException: Error while copying content to a stream.  ---> System.IO.IOException: Unable to write data to the transport connection: The I/O operation has been aborted because of either a thread exit or an application request..  ---> System.Net.Sockets.SocketException (995): The I/O operation has been aborted because of either a thread exit or an application request.    --- End of inner exception stack trace ---    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)    at System.Net.Security.SslStream.<WriteSingleChunk>g__CompleteWriteAsync|153_1[TIOAdapter](ValueTask writeTask, Byte[] bufferToReturn)    at System.Net.Security.SslStream.WriteAsyncChunked[TIOAdapter](ReadOnlyMemory`1 buffer, CancellationToken cancellationToken)    at System.Net.Security.SslStream.WriteAsyncInternal[TIOAdapter](ReadOnlyMemory`1 buffer, CancellationToken cancellationToken)    at System.Net.Http.HttpConnection.WriteAsync(ReadOnlyMemory`1 source, Boolean async)    at System.Net.Http.HttpContent.<CopyToAsync>g__WaitAsync|56_0(ValueTask copyTask)    --- End of inner exception stack trace ---    at System.Net.Http.HttpContent.<CopyToAsync>g__WaitAsync|56_0(ValueTask copyTask)    at System.Net.Http.HttpConnection.SendRequestContentAsync(HttpRequestMessage request, HttpContentWriteStream stream, Boolean async, CancellationToken cancellationToken)    at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)    --- End of inner exception stack trace ---    at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)    at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)    at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)    at Microsoft.Extensions.Http.Logging.LoggingHttpMessageHandler.<SendCoreAsync>g__Core|5_0(HttpRequestMessage request, Boolean useAsync, CancellationToken cancellationToken)    at Microsoft.Extensions.Http.Logging.LoggingScopeHttpMessageHandler.<SendCoreAsync>g__Core|5_0(HttpRequestMessage request, Boolean useAsync, CancellationToken cancellationToken)    at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)    --- End of inner exception stack trace ---    --- End of inner exception stack trace ---    at System.Net.Http.HttpClient.HandleFailure(Exception e, Boolean telemetryStarted, HttpResponseMessage response, CancellationTokenSource cts, CancellationToken cancellationToken, CancellationTokenSource pendingRequestsCts)    at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)    at WingetIntune.Implementations.ChunkedAzureFileUploader.UploadChunkAsync(String chunkId, Byte[] chunkData, Uri sasUri, CancellationToken cancellationToken) in /home/runner/work/WingetIntune/WingetIntune/src/WingetIntune/Implementations/ChunkedAzureFileUploader.cs:line 80    at WingetIntune.Implementations.ChunkedAzureFileUploader.UploadFileToAzureAsync(String filename, Uri sasUri, CancellationToken cancellationToken) in /home/runner/work/WingetIntune/WingetIntune/src/WingetIntune/Implementations/ChunkedAzureFileUploader.cs:line 55
fail: WingetIntune.IntuneManager[0] Error publishing app, deleting the remains System.Threading.Tasks.TaskCanceledException: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.  ---> System.TimeoutException: The operation was canceled.  ---> System.Threading.Tasks.TaskCanceledException: The operation was canceled.  ---> System.Net.Http.HttpRequestException: Error while copying content to a stream.  ---> System.IO.IOException: Unable to write data to the transport connection: The I/O operation has been aborted because of either a thread exit or an application request..  ---> System.Net.Sockets.SocketException (995): The I/O operation has been aborted because of either a thread exit or an application request.    --- End of inner exception stack trace ---    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)    at System.Net.Security.SslStream.<WriteSingleChunk>g__CompleteWriteAsync|153_1[TIOAdapter](ValueTask writeTask, Byte[] bufferToReturn)    at System.Net.Security.SslStream.WriteAsyncChunked[TIOAdapter](ReadOnlyMemory`1 buffer, CancellationToken cancellationToken)    at System.Net.Security.SslStream.WriteAsyncInternal[TIOAdapter](ReadOnlyMemory`1 buffer, CancellationToken cancellationToken)    at System.Net.Http.HttpConnection.WriteAsync(ReadOnlyMemory`1 source, Boolean async)    at System.Net.Http.HttpContent.<CopyToAsync>g__WaitAsync|56_0(ValueTask copyTask)    --- End of inner exception stack trace ---    at System.Net.Http.HttpContent.<CopyToAsync>g__WaitAsync|56_0(ValueTask copyTask)    at System.Net.Http.HttpConnection.SendRequestContentAsync(HttpRequestMessage request, HttpContentWriteStream stream, Boolean async, CancellationToken cancellationToken)    at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)    --- End of inner exception stack trace ---    at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)    at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)    at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)    at Microsoft.Extensions.Http.Logging.LoggingHttpMessageHandler.<SendCoreAsync>g__Core|5_0(HttpRequestMessage request, Boolean useAsync, CancellationToken cancellationToken)    at Microsoft.Extensions.Http.Logging.LoggingScopeHttpMessageHandler.<SendCoreAsync>g__Core|5_0(HttpRequestMessage request, Boolean useAsync, CancellationToken cancellationToken)    at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)    --- End of inner exception stack trace ---    --- End of inner exception stack trace ---    at System.Net.Http.HttpClient.HandleFailure(Exception e, Boolean telemetryStarted, HttpResponseMessage response, CancellationTokenSource cts, CancellationToken cancellationToken, CancellationTokenSource pendingRequestsCts)    at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)    at WingetIntune.Implementations.ChunkedAzureFileUploader.UploadChunkAsync(String chunkId, Byte[] chunkData, Uri sasUri, CancellationToken cancellationToken) in /home/runner/work/WingetIntune/WingetIntune/src/WingetIntune/Implementations/ChunkedAzureFileUploader.cs:line 80    at WingetIntune.Implementations.ChunkedAzureFileUploader.UploadFileToAzureAsync(String filename, Uri sasUri, CancellationToken cancellationToken) in /home/runner/work/WingetIntune/WingetIntune/src/WingetIntune/Implementations/ChunkedAzureFileUploader.cs:line 55    at WingetIntune.IntuneManager.AddContentVersionToApp(GraphServiceClient graphServiceClient, String appId, String intuneFilePath, CancellationToken cancellationToken) in /home/runner/work/WingetIntune/WingetIntune/src/WingetIntune/Intune/IntuneManager.cs:line 341    at WingetIntune.IntuneManager.PublishAppAsync(String packagesFolder, PackageInfo packageInfo, IntunePublishOptions options, CancellationToken cancellationToken) in /home/runner/work/WingetIntune/WingetIntune/src/WingetIntune/Intune/IntuneManager.cs:line 221
